pkg/kubectl/cmd/util/helpers.go

```diff
@@ -62,16 +63,40 @@ func AddSourceToErr(verb string, source string, err error) error {
 	return err
 }
 
+var fatalErrHandler = fatal
```

Comment:
The run methods should return errors.  If you want to test commands, run
them as sub shells.

On Wed, Sep 9, 2015 at 8:41 AM, Michal Fojtik <notifications@github.com>
wrote:

> In pkg/kubectl/cmd/util/helpers.go
> <https://github.com/kubernetes/kubernetes/pull/13728#discussion_r39036711>
> :
>
> > @@ -62,16 +63,40 @@ func AddSourceToErr(verb string, source string, err error) error {
> >  	return err
> >  }
> >
> > +var fatalErrHandler = fatal
>
> Using an env var will limit the possible functions that I can plug in
> here. For example if I want to panic() but before I want to collect some
> metrics or I want to log the error, etc. Maybe we can do both, support some
> 'pre-defined' functions (panic, fatal, etc..) but also allow to plug in
> completely custom function.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/13728/files#r39036711>.
>



-- 
Clayton Coleman | Lead Engineer, OpenShift